Filter for line items in abstract-wc-order.php

urosmil opened the pull request:
When creating custom product class there are places where line items is only value that can be handled. This is one of that places causing subtotal values after checkout not to be calculated and displayed.

urosmil committed:
Filter for line items in abstract-wc-order.php

When creating custom product class there are places where line items is only value that can be handled. This is one of that places causing subtotal values after checkout not to be calculated and displayed.

mikejolley commented:
Wouldn't filtering on `woocommerce_order_get_items` be a catch-all solution? We may need to add `$types` to the filter however so you know what is being requested.

urosmil committed:
Extension to woocommerce_order_get_items filter

This is extension to filter as get_items() is used for different classes and types. This way it would be possible to distinct which type is called for items.

urosmil commented:
Problem with `woocommerce_order_get_items ` is that WC_Order function uses get_items() not only for product classes but for shipping and others also. Solution with this filter would be to extend filter with types being used so inside function that is hooking to filter you can check which types are currently being used

mikejolley commented:
@urosmil If your PR added types to the filter, you could execute for line items only.

mikejolley commented on includes/abstracts/abstract-wc-order.php:
```diff
@@ -1599,9 +1599,9 @@ public function get_formatted_order_total() {
 		$tax_display = $tax_display ? $tax_display : get_option( 'woocommerce_tax_display_cart' );
 		$subtotal    = 0;
-
+		$items = apply_filters( 'woocommerce_subtotal_order_item_types', 'line_item' );
```
Bug here. Items will be a string, not an array of items.

urosmil replied:
@mikejolley If $types parameter is added as parameter for filter inside get_items() function than this one can be excluded.

urosmil committed:
Removed filtering inside subtotal function

$types parameter is added as parameter for filter inside get_items() function so previous commit is reverted as there is no need for it.
